Sources/TuistGenerator/Utils/CocoaPodsInteractor.swift

```diff
@@ -0,0 +1,102 @@
+import Foundation
+import TuistCore
+
+enum CocoaPodsInteractorError: FatalError, Equatable {
+    /// Thrown when CocoaPods cannot be found.
+    case cocoapodsNotFound
+
+    /// Error type.
+    var type: ErrorType {
+        switch self {
+        case .cocoapodsNotFound:
+            return .abort
+        }
+    }
+
+    /// Error description.
+    var description: String {
+        switch self {
+        case .cocoapodsNotFound:
+            return "CocoaPods was not found either in Bundler nor in the environment"
+        }
+    }
+}
+
+protocol CocoaPodsInteracting {
+    /// Runs 'pod install' for all the CocoaPods dependencies that have been indicated in the graph.
+    ///
+    /// - Parameter graph: Project graph.
+    /// - Throws: An error if the installation of the pods fails.
+    func install(graph: Graphing) throws
+}
+
+final class CocoaPodsInteractor: CocoaPodsInteracting {
+    /// Instance to output information to the user.
+    let printer: Printing
+
+    /// Instance to run commands in the system.
+    let system: Systeming
+
+    /// Initializes the CocoaPods
+    ///
+    /// - Parameters:
+    ///   - printer: Instance to output information to the user.
+    ///   - system: Instance to run commands in the system.
+    init(printer: Printing = Printer(),
+         system: Systeming = System()) {
+        self.printer = printer
+        self.system = system
+    }
+
+    /// Runs 'pod install' for all the CocoaPods dependencies that have been indicated in the graph.
+    ///
+    /// - Parameter graph: Project graph.
+    /// - Throws: An error if the installation of the pods fails.
+    func install(graph: Graphing) throws {
+        let canUseBundler = canUseCocoaPodsThroughBundler()
+        let canUseSystem = canUseSystemPod()
+
+        try graph.cocoapods.forEach { node in
+            var command: [String]
+
+            if canUseBundler {
+                command = ["bundle", "exec", "pod"]
+            } else if canUseSystem {
+                command = ["pod"]
+            } else {
+                throw CocoaPodsInteractorError.cocoapodsNotFound
+            }
+
+            command.append(contentsOf: ["install", "--project-directory=\(node.path.pathString)", "--repo-update"])
```

Comment:
it might be a good idea to pass through `--repo-update` since you might not always want to run this when running `tuist generate`

e.g. 

```
$ swift run tuist generate --path fixtures/ios_app_with_cocoapods -Xpod --repo-update

Reply:
Even better 😬. I've changed the implementation a bit:
- By default, it runs the command without the `--repo-update` argument.
- If CocoaPods fails because it can't find the version, it re-runs the command with the `--repo-update` argument.

I think the simpler we keep Tuist's interface, the better. If it fails because the repo of specs is outdated, I think Tuist should handle that automatically.